### PR TITLE
New/toolbar

### PR DESCRIPTION
**Pull request by ovchynnikov**

### CODE REVIEW

File: backend/src/data/repositories/page-repository.ts [hunk 1]
```diff
@@ -1,11 +1,19 @@
 import { EntityRepository, Repository } from 'typeorm';
 import { Page } from '../entities/page';
-
```
Comment (aleksandr-kuharenko) [inline backend/src/data/repositories/page-repository.ts:3]: This blank line needed here =)

File: backend/src/services/page.service.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { getCustomRepository } from 'typeorm';
+import PageRepository from '../data/repositories/page-repository';
+import UserRepository from '../data/repositories/user.repository';
+import { Page } from '../data/entities/page';
+import UserPermissionRepository from '../data/repositories/user-permission.repository';
+import TeamPermissionRepository from '../data/repositories/team-permission-repository';
+import { IRequestWithUser } from '../common/models/user/request-with-user.interface';
+
+export const getPages = async (req: IRequestWithUser): Promise<Page[]> => {
```
Comment (aleksandr-kuharenko) [inline backend/src/services/page.service.ts:9]: do not send `req` to service. Only needed payload

File: backend/src/services/page.service.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  const userTeamsPermissions = await teamPermissionRepository.findByTeamId(teamId);
+
+  const userPermissions = await userPermissionRepository.findById(userId);
+
+  const allPages = await pageRepository.findPages(workspaceId);
+
+  interface IPageWithChildren extends Page {
+    children?: Page[]
+  }
```
Comment (aleksandr-kuharenko) [inline backend/src/services/page.service.ts:29]: why interface is defined in body of service?

**Review by aleksandr-kuharenko — CHANGES_REQUESTED** [review]

File: backend/src/data/repositories/page-repository.ts [hunk 4]
```diff
@@ -1,11 +1,19 @@
...
 class PageRepository extends Repository<Page> {
   public findById(id: string):Promise<Page> {
     return this.findOne({ id });
   }
-}
 
+  findPages(workspaceId: string): Promise<Page[]> {
+    return this.find({
+      relations: ['pageContents'],
+      where: { workspaceId: workspaceId },
```
Comment (aleksandr-kuharenko) [inline backend/src/data/repositories/page-repository.ts:12]:
```suggestion
      where: { workspaceId },
```

File: frontend/src/components/toolbar/components/plus-button/plus-button.tsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const PlusButton: React.FC = () => {
+  return (
+    <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" className="bi bi-plus-lg ms-auto" viewBox="0 0 16 16">
+      <path d="M8 0a1 1 0 0 1 1 1v6h6a1 1 0 1 1 0 2H9v6a1 1 0 1 1-2 0V9H1a1 1 0 0 1 0-2h6V1a1 1 0 0 1 1-1z"/>
+    </svg>
+  );
+};
+
+export default PlusButton;
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/toolbar/components/plus-button/plus-button.tsx:9]: we use https://icons.getbootstrap.com/ for icons, check example in our codebase

File: shared/package.json [hunk 6]
```diff
@@ -1,20 +1,20 @@
...
+        "lint:js": "npx eslint \"src/**/*.ts\"",
+        "lint": "npm run lint:js",
+        "build:ts": "tsc && tscpaths -p tsconfig.json --src src --out build",
+        "build": "npm run build:ts && cp package.json build"
+    },
+    "devDependencies": {
+        "tscpaths": "0.0.9",
+        "typescript": "4.3.5"
+    },
+    "dependencies": {}
```
Comment (aleksandr-kuharenko) [inline shared/package.json:19]: revert these changes back

File: frontend/src/components/toolbar/toolbar.tsx [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+      <div className="pt-5">
+        <SectionName name="Pages" />
+        <Accordion className={styles.accordion} defaultActiveKey="1" flush>
+          <PagesList pages={pages.pages}/>
+          <p>below hardcoded</p>
+          <Accordion className={styles.accordion} flush>
+            <Accordion.Item eventKey="0" className={getAllowedClasses('pl-4',styles.accordionItem, styles.accordionItemInsideSection)}>
+              <Nav.Link className={getAllowedClasses(styles.navbarBrand, styles.navbarLinkInsideSection)}>Settings<PlusButton/></Nav.Link>
+            </Accordion.Item>
+          </Accordion>
```
Comment (aleksandr-kuharenko) [inline frontend/src/components/toolbar/toolbar.tsx:34]: remove all hard code

File: shared/package.json (hunk 6 above)
Comment (ovchynnikov) [reply shared/package.json:19]: can't make a revert, just copied the correct spacing from package.json file on develop branch. My file was changed by vscode

**Review by AndriiPopov3 — APPROVED** [review]

**Review by AlikRakhmonov — APPROVED** [review]

**Review by aleksandr-kuharenko — APPROVED** [review]

**Review by ursarik — APPROVED** [review]

**Pull request merged**